add human-panic for user-friendly error messages

OmarElhagagy opened the pull request:
Fixes: Use human-panic to improve error messages

- Added `human-panic` dependency to `Cargo.toml`
- Integrated `setup_panic!()` in `main.rs`

OmarElhagagy committed:
add human-panic for user-friendly error messages

2sugarcubes commented:
Thanks for contributing to the repository. I'll have a look at those failed workflows and get back to you with some advice to get it passing, or I can commit those changes to get tests passing myself if you prefer.

2sugarcubes commented on bin/src/main.rs:
```diff
@@ -20,9 +21,17 @@ use rand::{Rng, SeedableRng};
+    human_panic::setup_panic!();
 
+    if let Err(e) = try_main() {
+        eprintln!("Error: {}", e);
```
Clippy would like this formatting literal to contain the error variable like this `eprintln!("Error: {e}");`

Also, would you consider using [`log::error!`](https://docs.rs/log/0.4.25/log/macro.error.html) in place of `eprintln`? It's up to you but it would mean that all warnings/errors/logs/etc. are formatted in the same way across the binary.

2sugarcubes reviewed (changes requested):
Total nitpick, but see comment on [line 28](https://github.com/2sugarcubes/astrograph/pull/15/commits/edcfcac469a0fb948d1de535605e3796a19472f6#r1943513060)

2sugarcubes commented:
Sorry if I was curt. I just wanted to let you know that you did a good job with this PR and it is my fault that [line 28] wasn't caught in your IDE and instead in CI. I'll fix-up this PR and merge it if I don't hear back from you in 7 days (the 14th of Febuary where I am)

OmarElhagagy commented:
Thanks for the review! this is like my second time using rust so i could've dropped some concepts. I’ll update the eprintln! line to use log::error! and fix the formatting per Clippy’s suggestion. I’ll also double-check my local linting setup to catch these earlier in the future. Let me know if there’s anything else to adjust!

2sugarcubes commented:
Missing that lint is definitely on me, because the CI uses an explicit deny for `clippy::pedantic` which makes it catch a lot more lints than normal, along with *some* false positives. I'll add that to local files as well once this PR is merged.

OmarElhagagy committed:
fix: update error handling and string formatting

- Changed eprintln! to log::error!
- Updated string formatting to follow Clippy suggestions
- Fixed Clippy pedantic lints

2sugarcubes reviewed (approved):
Thanks for your work :+1:. I'll get on merging it now 🎊

2sugarcubes committed:
Merge branch 'dev' into human-panic-errors

2sugarcubes commented:
Don't worry about those workflow fails, that was my fault because of the merge. I'll fix it and merge.

2sugarcubes committed:
chore(binary): removed unneeded import

2sugarcubes commented:
Thank you again. I've merged your PR into the dev branch now, and you can see your commits on the [contributors page](https://github.com/2sugarcubes/astrograph/graphs/contributors)

OmarElhagagy commented:
Thanks for merging! Glad to contribute, and no worries about the workflow appreciate you fixing it.
